charms/istio-pilot/src/charm.py

```diff
@@ -181,23 +214,57 @@ def handle_ingress_auth(self, event):
 
         manifests = [rbac_configs, auth_filters]
         manifests = '\n'.join([m for m in manifests if m])
-        subprocess.run(
-            [
-                './kubectl',
-                'delete',
-                'envoyfilters,rbacconfigs',
-                f'-lapp.juju.is/created-by={self.model.app.name}',
-                '-n',
-                self.model.name,
-            ],
-            check=True,
-        )
 
-        subprocess.run(
-            ["./kubectl", "apply", "-f-"],
-            input=manifests.encode('utf-8'),
-            check=True,
-        )
+        for resource in [self.envoy_filter_resource, self.rbac_config_resource]:
+            self._delete_existing_resource_objects(resource, namespace=self.model.name)
+        self._apply_manifest(manifests, namespace=self.model.name)
+
+    def _delete_object(
+        self, obj, namespace=None, ignore_not_found=False, ignore_unauthorized=False
+    ):
+        try:
+            self.lightkube_client.delete(obj.__class__, obj.metadata.name, namespace=namespace)
+        except ApiError as err:
+            self.log.error("ApiError encountered while attempting to delete resource.")
+            if err.status.message is not None:
+                if "not found" in err.status.message and ignore_not_found:
+                    self.log.error(f"Ignoring not found error:\n{err.status.message}")
+                elif "(Unauthorized)" in err.status.message and ignore_unauthorized:
+                    # Ignore error from https://bugs.launchpad.net/juju/+bug/1941655
+                    self.log.error(f"Ignoring unauthorized error:\n{err.status.message}")
+                else:
+                    self.log.error(err.status.message)
+                    raise
+            else:
+                raise
+
+    def _delete_existing_resource_objects(
+        self, resource, namespace=None, ignore_not_found=False, ignore_unauthorized=False
+    ):
+        for obj in self.lightkube_client.list(
+            resource, labels={"app.juju.is/created-by": f"{self.app.name}"}
+        ):
+            self._delete_object(
+                obj,
```

Comment:
is this `obj` or is `type(obj)`.  If it really is meant to be `klass` -- i suggest a rename here in this loop

but when I [`read-the-docs`](https://lightkube.readthedocs.io/en/latest/client/#list-or-watch-objects-in-all-namespaces) it seems `client.list(...)` does return the entire object.  In other circumstances you've used `type(obj)` during deletions.  Can you verify this is doing the right thing?

Reply:
so the lightkube delete method takes a class as its first argument, the resource kind. As its second argument it takes the name of the object that is to be deleted. This felt weird to me so I added the _delete_object helper method that just takes in an object, which seemed less confusing than having to pass in a resource type and object name explicitly.

Reply:
ooo!  i see this now.  thanks for clearing that up.